src/protocol/clock/period.rs

```diff
@@ -0,0 +1,108 @@
+use std::num::{IntErrorKind, TryFromIntError};
+use std::time::Duration;
+
+pub type PeriodLength = Duration;
+pub type PeriodCount = u64;
```

Comment:
Does `PeriodCount` have to be a `u64`? A `u32` would be a lot more convenient as argument for `checked_mul`, `checked_add`, etc..

Reply:
If the increment is less than one second, you can easily overflow the u32 size, in this case u64 is better.